src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let ACTION_REQUIRED = '';
+  const stateCopy = { ...state };
+  const ARRAY = [Object.assign({}, stateCopy)];
+
+  for (const action of actions) {
+    ACTION_REQUIRED = action.type;
+    let elementArray = Object.assign({}, ARRAY[ARRAY.length - 1]);
+
+    switch (ACTION_REQUIRED) {
+      case 'addProperties':
+        elementArray = Object.assign(elementArray, action.extraData)
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete elementArray[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in elementArray) {
+          delete elementArray[key];
+        };
+        break;
+    }
+
+    ARRAY.push(elementArray);
+  }
+  ARRAY.shift();
```

Comment:
ARRAY.shift() removes the first element of the array which is not necessary, remove this line.

Reply:
Without ARRAY.shift() the tests aren't passed. The first element in the array is state{} and ARRAY.shift() remove it.